src/refract/primitives.es6

```diff
@@ -0,0 +1,322 @@
+/*
+ * A refract element implementation with an extensible type registry.
```

Comment:
> Should ArrayType().push take in refracted type instances or plain js values?
> Should ObjectType().set take in refracted type instances or plain js values?

We could probably allow for both in some way, I'd say. Can we do this? In other words, if the type received is an instance, just store it, otherwise, convert it. Or have some options argument for `push`, `set`, etc. that allows you say not convert the value argument.